cdap-api/src/main/java/co/cask/cdap/api/TransactionUtil.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.api;
+
+import co.cask.cdap.api.data.DatasetContext;
+import org.apache.tephra.TransactionFailureException;
+
+import java.util.concurrent.atomic.AtomicReference;
+import javax.annotation.Nullable;
+
+/**
+ * Helper class for interacting with {@link Transactional}
+ * This is similar to co.cask.cdap.data2.transaction.Transactions, but added here to avoid dependency
```

Comment:
Don't add this. This is public API. The Javadoc shouldn't mention any internal class.